test/sys/test_event.rs

```diff
@@ -0,0 +1,30 @@
+use nix::sys::event;
+use libc;
+use std::mem::{size_of, size_of_val};
+use std::ptr;
+
+#[test]
+/// Check that nix::event::KEvent has the same layout as libc::kevent
+/// This would be easier with offsetof operators.
```

Comment:
if we do go this route, then the test should set distinct non-zero values in the fields and also check for equality.